### PR TITLE
MAINT-48685: Catch CsrfException and reload the page when the csrf token has been lost or expired

### PR DESCRIPTION
**Pull request by hakermi**
…
ISSUE: When the csrf token is lost and expired in the session, and we have alread opened pages of the doc app, This lead to a blank page when execute any action that demands a csrf token check
FIX: Reload the page when a csrf token is expired or lost to avoid the display of blank page and to refresh the token in the page

### CODE REVIEW

File: webui/framework/src/main/java/org/exoplatform/webui/core/UIApplication.java [hunk 1]
```diff
@@ -123,10 +124,12 @@ public void processAction(WebuiRequestContext context) throws Exception {
             super.processAction(context);
         } catch (MessageException ex) {
             addMessage(ex.getDetailMessage());
-        } catch (Throwable t) {
+        } catch (CSRFException e) {
+            context.getJavascriptManager().getRequireJS().addScripts("location.reload();");
+        } catch (Exception e) {
```
Comment (rdenarie) [inline webui/framework/src/main/java/org/exoplatform/webui/core/UIApplication.java]:
```suggestion
        } catch (Trhowable t) {
```

File: webui/framework/src/main/java/org/exoplatform/webui/core/UIApplication.java [hunk 2]
```diff
@@ -123,10 +124,12 @@ public void processAction(WebuiRequestContext context) throws Exception {
...
             ApplicationMessage msg = new ApplicationMessage("UIApplication.msg.unknown-error", null, ApplicationMessage.ERROR);
             addMessage(msg);
-            log.error("Error during the processAction phase", t);
+            log.error("Error during the processAction phase", e);
```
Comment (rdenarie) [inline webui/framework/src/main/java/org/exoplatform/webui/core/UIApplication.java]:
```suggestion
            log.error("Error during the processAction phase", t);
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: webui/framework/src/main/java/org/exoplatform/webui/exception/CSRFException.java [hunk 3]
```diff
@@ -1,6 +1,6 @@
 package org.exoplatform.webui.exception;
 
-public class CSRFException extends Exception{
+public class CSRFException extends RuntimeException{
```
Comment (rdenarie) [inline webui/framework/src/main/java/org/exoplatform/webui/exception/CSRFException.java]: Why did you change the exception type ? Is there a particular reason ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: webui/framework/src/main/java/org/exoplatform/webui/exception/CSRFException.java (hunk 3 above)
Comment (hakermi) [reply webui/framework/src/main/java/org/exoplatform/webui/exception/CSRFException.java]: There is no major difference, but **RuntimeException** is an unchecked exception, and those exception are used when there is no recover for the thrown issue which is dealing with our situation where we cant' do anything except reloading the page when this exception comes to happen

Comment (hakermi) [reply webui/framework/src/main/java/org/exoplatform/webui/exception/CSRFException.java]: After thinking, I restore the Exception as we are handling it in a try catch clause

**Review by rdenarie — APPROVED** [review]
